src/aws_credentials_httpc.erl

```diff
@@ -10,20 +10,29 @@
 %% @end
 
 -module(aws_credentials_httpc).
--export([start/0, get/1, get/2]).
+-export([start/0, get/1, get/2, get/3, put/1, put/2, put/3]).
 
 -define(PROFILE, aws_credentials).
 -define(TIMEOUT, 10000). % 10 sec
 -define(CONNECT_TIMEOUT, 3000). % 3 sec
 -define(DEFAULT_TRIES, 3).
+-define(DEFAULT_HEADERS, []).
 -define(DELAY, 100). % 100 microseconds
 
+-type field() :: string().
+-type value() :: string().
+-type request_header() :: {field(), value()}.
+-type request_headers() :: [request_header()].
+
 start() ->
     inets:start(httpc, [{profile, ?PROFILE}]).
 
 %% @doc Attempt to get a URL with the 3 retries. 3 is the default.
 get(URL) ->
-  get(URL, ?DEFAULT_TRIES).
+  get(URL, ?DEFAULT_HEADERS, ?DEFAULT_TRIES).
+
+get(URL, RequestHeaders) ->
```

Comment:
I changed get to allow request headers, and then essentially duplicated the `get` code for `put`.  It seems like I should combine these but thought I'd get feedback first.

Reply:
Ideally you'd have a `request` function that takes the method `get` or `put` as a parameter